### PR TITLE
gnirehtet: update cargo.lock to fix build with rust 1.64.0

### PR DESCRIPTION
**Pull request by cho-m**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Needs new bottles since Linux is hitting relocation issue:
```
  ==> Pouring gnirehtet--2.5_1.x86_64_linux.bottle.tar.gz
  Error: cannot normalize PT_NOTE segment: non-contiguous SHT_NOTE sections
```

However, rebottle attempts fails with Rust 1.64.0+ - https://github.com/Homebrew/homebrew-core/actions/runs/3589552605/jobs/6042109311

### CODE REVIEW

File: Formula/gnirehtet.rb [hunk 1]
```diff
@@ -29,6 +30,12 @@ class Gnirehtet < Formula
   def install
     resource("java_bundle").stage { libexec.install "gnirehtet.apk" }
 
+    # Building the binary with rust 1.64.0 or later results in an error when running `gnirehtet relay`.
+    # ERROR Main: Execution error: IO error: Address family not supported by protocol family (os error 47)
+    # Issue ref: https://github.com/Genymobile/gnirehtet/issues/475
+    system "rustup-init", "-qy", "--no-modify-path", "--default-toolchain", "1.63.0"
+    ENV.prepend_path "PATH", HOMEBREW_CACHE/"cargo_cache/bin"
```
Comment (cho-m) [inline Formula/gnirehtet.rb:37]: If anyone knows of an upstreamable fix for Rust 1.64.0+, then we can patch it in instead.

For now, went with workaround of downgrading Rust version.

**Review by carlocab — DISMISSED** [review]
What version are our bottles built with? Is the error from building with the wrong version of Rust caught by the test?

**Comment by cho-m** [comment]
> What version are our bottles built with?

A mix of versions, e.g.
- Monterey bottles were built in https://github.com/Homebrew/homebrew-core/commit/2b185d95930b265a444283cb7837195854eaa0dc with Rust 1.56.0
- Linux bottle was built in https://github.com/Homebrew/homebrew-core/commit/9b2420c2952c69b05cb20ad476365a4c42b2d2a0 with Rust 1.53.0
- Catalina and older bottles were built in https://github.com/Homebrew/homebrew-core/commit/e2b29984257fadd6adf2cc78291a13ee90b179ab with Rust 1.45.2


> Is the error from building with the wrong version of Rust caught by the test?

Yes, test failed in Rust 1.64.0 PR #111434 - https://github.com/Homebrew/homebrew-core/actions/runs/3109704394/jobs/5041504928#step:10:4266

**Review by chenrui333 — DISMISSED** [review]

**Comment by chenrui333** [comment]
> Last commit in 2020, lots of open trivial PRs I'd just mark it as an unmaintained project

Raised an issue in the upstream about the project status, we can revisit maybe couple of months later.

**Comment by cho-m** [comment]
Upstream has created a PR to update Rust dependencies in Cargo.lock to fix issue. We can try the PR commit instead.

**Comment by chenrui333** [comment]
Agree, let's try the upstream PR patch instead.

File: Formula/gnirehtet.rb [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
   license "Apache-2.0"
   revision 1
   head "https://github.com/Genymobile/gnirehtet.git", branch: "master"
 
+  stable do
+    url "https://github.com/Genymobile/gnirehtet/archive/v2.5.tar.gz"
+    sha256 "2b55b56e1b21d1b609a0899fe85d1f311120bb12b04761ec586187338daf6ec5"
+
+    # Fix compilation issue with rust 1.64.0+
+    # upstream PR reference, https://github.com/Genymobile/gnirehtet/pull/478
```
Comment (chenrui333) [inline Formula/gnirehtet.rb:13]: The direct `git apply` has some conflict, I just re-generate the cargo.lock from 2.5 release artifact.

**Review by chenrui333 — APPROVED** [review]

**Pull request closed without merging**